vignettes/site_only/deploy_model.Rmd

```diff
@@ -0,0 +1,189 @@
+- We read the model from disk; if needed, a web service could be involved
+
+
+```{r}
+prod_query <- "SELECT 
+                 [FacilityAccountID]
+                ,[AdmitAgeNBR]
+                ,[GenderCD]
+                ,[EthnicGroupDSC]
+                ,[MaritalStatusDSC]
+                ,[Readmit30FLG]
+  FROM [SAM].[Schema].[SummaryVisitBASE]
+  WHERE InpatientFLG = 'T'"
+
+d_out <- db_read(con, prod_query)
+
+load("readmit_models.RDA")
+
+predictions <- predict(models, d_out)
+```
+
+## Create the output table
+
+Before we create the output table, look at the format of your output by typing `glimpse(predictions)`
+
+You might want push fewer columns to a table each night. Here's how you slim this down:
+
+```{r}
+predictions <- predictions %>% 
+  select(FacilityAccountID, predicted_Readmit30FLG) %>% # Grab just two columns
+  rename(PredictedProbability = predicted_Readmit30FLG) # Change col name
+```
+
+Now that we have only the columns necessary, let's and populate the output table
+
+### Outside of a Health Catalyst environment
+
+Use a tool like SQL Server Management Studio and a `CREATE TABLE` statement
+
+### Inside a Health Catalyst environment
+
+1. Append a few utility columns to your output
+
+```{r}
+predictions <- predictions %>%
+  add_SAM_utility_cols
+```
+
+2. Use Source Area Mart Designer (SAMD) to create the table (since this creates metadata)
+  - This binding requires a query that not only has the column structure, but also `WHERE 0 = 1`, such that only R is populating the output table
+
+## Writing predictions to the table
+
+Here we use the `RODBC` package, which can be installed via `install.packages('RODBC')`
+
+```{r}
+library(RODBC)
+
+con <- odbcDriverConnect(connection = my_con)
+
+sqlSave(con, 
+        predictions, 
+        "Schema.MLOutputTable", 
+        append = TRUE, 
+        rownames = FALSE)
+
+odbcClose(con)
+```
+
+If your predictions arrive in the database, congrats! You're now ready to work with the DBA/ETL* team to schedule this reoccurring job, such that it runs in a manner that's dependency-aware.
+
+# Testing in production
+
+Before users start consuming your awesome ML work, it's imperative that you verify that the model's accuracy holds up in production. Performance on retrospective data isn't the same as performance on incoming production data. Sometimes issues like [data leakage](https://healthcare.ai/data-leakage-in-healthcare-machine-learning) or other issues cause the performance to drop.
+
+Effectively, if you're creating CLABSI predictions, you should wait a few weeks until a large number of the patients who have received risk predictions have left the hospital. Alternatively, if you're predicting 30-day readmissions, you need to wait ~45 days until you have enough data to verify if you accurately predicted who did or didn't return to the hospital. This can be done [via healthcare.ai](https://docs.healthcare.ai/reference/evaluate_classification.html):
```

Comment:
~45 days may be sufficient run-out time for 30-day readmissions. As I think about it harder though, it might need to be volume and class balance and prediction timing specific. I mention volume because for a really specific model, there may only be a handful of discharges per day; class balance because the outcome may be so rare (like CLABSI or an orthopedic related readmission) that a few weeks or even 45 days may not have any positive cases (might be fine); prediction timing because some readmission models may continue to make predictions after discharge (providing 45+ days of volume) which would increase volume, while others may stop at discharge (meaning that you'll only have 15 days of volume to evaluate the model).